spec/core/ics-002-client-semantics/README.md

```diff
@@ -196,95 +255,64 @@ A validity predicate is an opaque function defined by a client type to verify `H
+`checkForMisbehaviour` MUST throw an exception if the provided proof of misbehaviour was not valid.
 
-Once misbehaviour is detected, clients SHOULD be frozen so that no future updates can be submitted.
-A permissioned entity such as a chain governance system or trusted multi-signature MAY be allowed
-to intervene to unfreeze a frozen client & provide a new correct header.
-
-#### Height
+#### UpdateState
 
-`Height` is an opaque data structure defined by a client type.
-It must form a partially ordered set & provide operations for comparison.
-
-```typescript
-type Height
-```
+UpdateState will update the client given a verified `ClientMessage`. Note that this function is intended for **non-misbehaviour** `ClientMessages`.
 
 ```typescript
-enum Ord {
-  LT
-  EQ
-  GT
-}
-
-type compare = (h1: Height, h2: Height) => Ord
+type UpdateState = (ClientMessage) => Void
 ```
 
-A height is either `LT` (less than), `EQ` (equal to), or `GT` (greater than) another height.
-
-`>=`, `>`, `===`, `<`, `<=` are defined through the rest of this specification as aliases to `compare`.
+`verifyClientMessage` must be called before this function, and `checkForMisbehaviour` must return false before this function is called.
 
-There must also be a zero-element for a height type, referred to as `0`, which is less than all non-zero heights.
+The client MUST also mutate internal state to store
+now-finalised consensus roots and update any necessary signature authority tracking (e.g.
+changes to the validator set) for future calls to the validity predicate.
 
-#### ClientState
+Clients MAY have time-sensitive validity predicates, such that if no ClientMessage is provided for a period of time
+(e.g. an unbonding period of three weeks) it will no longer be possible to update the client.
+In this case, a permissioned entity such as a chain governance system or trusted multi-signature MAY be allowed
+to intervene to unfreeze a frozen client & provide a new correct ClientMessage.
 
-`ClientState` is an opaque data structure defined by a client type.
-It may keep arbitrary internal state to track verified roots and past misbehaviours.
+#### UpdateStateOnMisbehaviour
 
-Light clients are representation-opaque — different consensus algorithms can define different light client update algorithms —
-but they must expose this common set of query functions to the IBC handler.
+UpdateStateOnMisbehaviour will update the client upon receiving a verified `ClientMessage` that is valid misbehaviour.
 
 ```typescript
-type ClientState = bytes
+type UpdateStateOnMisbehaviour = (ClientMessage) => Void
 ```
 
-Client types MUST define a method to initialise a client state with a provided consensus state, writing to internal state as appropriate.
-
-```typescript
-type initialise = (consensusState: ConsensusState) => ClientState
-```
+`verifyClientMessage` must be called before this function, and `checkForMisbehaviour` must return `true` before this function is called.
```

Comment:
This is confusion: Does `UpdateStateOnMisbehaviour` require both `verifyClientMessage` and `checkForMisbehaviour` to be called?